src/main/java/build/buildfarm/common/blake3/JniLoader.java

```diff
@@ -12,63 +12,50 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package build.buildfarm.common.jni;
+package build.buildfarm.common.blake3;
 
 import static com.google.common.base.Preconditions.checkArgument;
 
 import com.google.common.io.ByteStreams;
-import com.google.devtools.build.lib.util.OS;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.logging.Level;
+import javax.annotation.Nullable;
+import lombok.extern.java.Log;
 
 /** Generic code to interact with the platform-specific JNI code bundle. */
+@Log
 public final class JniLoader {
+  @Nullable private static final Throwable JNI_LOAD_ERROR;
 
-  private static final boolean JNI_AVAILABLE;
+  private static boolean isDarwin() {
+    return System.getProperty("os.name").equals("Mac OS X");
+  }
+
+  private static boolean isWindows() {
+    return System.getProperty("os.name").startsWith("Windows");
+  }
 
   static {
-    boolean jniAvailable;
+    Throwable jniLoadError;
     try {
-      switch (OS.getCurrent()) {
-        case LINUX:
-        case FREEBSD:
-        case OPENBSD:
-        case UNKNOWN:
-          loadLibrary("main/native/libunix_jni.so");
-          break;
-
-        case DARWIN:
-          loadLibrary("main/native/libunix_jni.dylib");
-          break;
-
-        case WINDOWS:
-          try {
-            // TODO(jmmv): This is here only for the bootstrapping process, which builds the JNI
-            // library and passes a -Djava.library.path to the JVM to find it. I'm sure that this
-            // can be replaced by properly bundling the library as a resource in the JAR. For some
-            // strange reason that I haven't fully understood yet, this also must come first.
-            System.loadLibrary("windows_jni");
-          } catch (UnsatisfiedLinkError e) {
-            try {
-              loadLibrary("main/native/windows/windows_jni.dll");
-            } catch (IOException e2) {
-              throw e;
-            }
-          }
-          break;
-
-        default:
-          throw new AssertionError("switch statement out of sync with OS values");
+      if (isDarwin()) {
+        loadLibrary("main/native/libblake3_jni.dylib");
```

Comment:
gonna assume even the name is diff, what's behind the dylib is probably the same as `main/native/libunix_jni.dylib`?

Reply:
Yes, it's the same